src/v/random/generators.h

```diff
@@ -33,8 +33,24 @@ static thread_local std::default_random_engine gen(internal::get_seed());
 } // namespace internal
 
 bytes get_bytes(size_t n = 128 * 1024);
+
+/**
+ * Random string generator. Total number of distinct values that may be
+ * generated is unlimited (within all possible values of given size).
+ */
 ss::sstring gen_alphanum_string(size_t n);
 
+static constexpr size_t alphanum_max_distinct_strlen = 32;
```

Comment:
nit: as i understand `gen_alphanum_max_distinct` returns random string with fixed length equal to  `alphanum_max_distinct_strlen`; because the string length is fixed it's better to rename it to `alphanum_distinct_strlen`